brjs-sdk/sdk/libs/javascript/br-i18n/src/br/i18n/Translator.js

```diff
@@ -46,7 +46,7 @@ function Translator(messageDefinitions, useLocale) {
 
 	/** @private */
 	this.localizationPrefs = {};
-	this.setLocale(useLocale);
+	this.setLocale(useLocale || 'en');
```

Comment:
If there is no locale defined in  `useLocale` and `en` doesn't exist then there might be issues but this shouldn't be possible in a BRJS app, I believe it expects a locale to be defined?

From my reading of the code `defaultLocale` is badly named. It is in fact a `productionFallbackLocale` as it's a fallback that only works in production

```javascript
var message = this.messages[token];
if (typeof message === 'undefined') {
	log.warn(Translator.MESSAGES.UNTRANSLATED_TOKEN_LOG_MSG, token, this.locale);
	if (!require('service!br.app-meta-service').isDev()) {
		message = this.defaultLocaleMessages[token];
	}
}
```

If we were to use the fallback the UTs will still fail as it only works in prod, the `Translator` would return the i18n token key and not the translated value which is what the tests in mobile expect. As you say yourself this change is for when no locales have been defined, in prod you would expect a locale to be defined.